templates/ModelDecoderDateProperty.stencil

```diff
@@ -1,7 +1,12 @@
 if let dateString = try? container.decode(String.self, forKey: .{{ property.name }}) {
     let dateFormatter = DateFormatter()
     dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSZ"
-    self.{{ property.name }} = dateFormatter.date(from: dateString)
+    if let date = dateFormatter.date(from: dateString) {
+        self.{{ property.name }} = date
+    } else {
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        self.{{ property.name }} = dateFormatter.date(from: dateString)
+    }
```

Comment:
Perhaps we just has a list of date formats 
```Swift
let formats = [
  "yyyy-MM-dd'T'HH:mm:ss.SSSSZ",
  "yyyy-MM-dd'T'HH:mm:ssZ"
]
```

And then we just iterate through those until one succeeds, and then break. That way it would be easier to add other odd formats.